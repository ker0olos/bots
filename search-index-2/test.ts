// deno-lint-ignore no-external-import
import { create, insert, search } from 'npm:@orama/orama';

import utils from '~/src/utils.ts';

import { CharactersDirectory } from '~/src/packs.ts';

// const _loadedAnilistCharactersDirectory: CharactersDirectory = await utils
//   .readJson<
//     CharactersDirectory
//   >('./packs/anilist/characters_directory.json');

// const map = new Map<string, boolean>();

const db = await create({
  schema: {
    name: 'string[]',
    mediaTitle: 'string[]',
    popularity: 'number',
  },
});

// await Promise.all(_loadedAnilistCharactersDirectory.map(async (doc) => {
//   if (!map.has(doc.id)) {
//     map.set(doc.id, true);
//     return await insert(db, doc as never);
//   }
// }));

// Deno.writeTextFile('./test.json', JSON.stringify(db));

const dbLoaded = await utils
  .readJson('./test.json');

//@ts-ignore
db.data = dbLoaded.data;
//@ts-ignore
db.caches = dbLoaded.caches;
//@ts-ignore
db.internalDocumentIDStore = dbLoaded.internalDocumentIDStore;

const searchResult = await search(db, {
  term: 'luka',
});

console.log(searchResult.hits.map((hit) => hit.document));
