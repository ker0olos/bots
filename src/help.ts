import i18n from '~/src/i18n.ts';
import user from '~/src/user.ts';

import * as discord from '~/src/discord.ts';

import {
  COSTS,
  MAX_PULLS,
  RECHARGE_DAILY_TOKENS_HOURS,
  RECHARGE_MINS,
  STEAL_COOLDOWN_HOURS,
} from '~/db/mod.ts';

import { PARTY_PROTECTION_PERIOD } from '~/src/steal.ts';
import { RECHARGE_KEYS_MINS } from '~/db/mod.ts';

function pages(
  { userId, index }: { userId: string; index: number },
): discord.Message {
  const locale = user.cachedUsers[userId]?.locale;

  const pages = [
    new discord.Message()
      .addComponents([
        new discord.Component()
          .setId('gacha', userId)
          .setLabel('/gacha'),
        new discord.Component()
          .setId('now', userId)
          .setLabel('/now'),
      ])
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: '1.' })
          .setTitle('`/gacha`')
          .setDescription([
            i18n.get('help-page1-before-anything', locale),
            '',
            i18n.get(
              'help-page1-you-get-n-every-t',
              locale,
              RECHARGE_MINS,
              MAX_PULLS,
            ),
            '',
            i18n.get('help-page1-characters-are-exclusive', locale),
            '',
            i18n.get(
              'help-page1-characters-vary',
              locale,
              discord.emotes.smolStar,
            ),
            '',
            i18n.get('help-page1-gacha-rates', locale, discord.emotes.smolStar),
            '',
            i18n.get('help-page1-footer', locale),
            discord.empty,
          ].join('\n'))
          .setFooter({ text: `${i18n.get('aliases', locale)}: /w, /q` }),
      ),
    new discord.Message()
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: '2.' })
          .setTitle('`/merge`')
          .setDescription([
            i18n.get('help-page2-sacrifice-characters', locale),
            '',
            i18n.get('help-page2-merge-protect', locale),
            '',
            i18n.get('help-page2-merging', locale),
            '',
            i18n.get('help-page2-auto-merging', locale),
            '',
            i18n.get(
              'help-page2-merge-example-1',
              locale,
              discord.emotes.smolStar,
            ),
            i18n.get(
              'help-page2-merge-example-2',
              locale,
              discord.emotes.smolStar,
            ),
            '',
            i18n.get('help-page2-footer', locale),
            discord.empty,
          ].join('\n'))
          .setFooter({ text: `${i18n.get('aliases', locale)}: /synthesis` }),
      ),
    new discord.Message()
      .addComponents([
        new discord.Component()
          .setId('cstars', '5', userId)
          .setLabel('/collection stars 5'),
        new discord.Component()
          .setId('cstars', '4', userId)
          .setLabel('/collection stars 4'),
      ])
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: '3.' })
          .setTitle('`/party`')
          .setDescription(
            [
              i18n.get('help-page3-party-required', locale),
              '',
              i18n.get('help-page3-party-presets-disclaimer', locale),
              '',
              i18n.get('help-page3-party-view', locale),
              '',
              i18n.get('help-page3-party-assign-1', locale),
              i18n.get('help-page3-party-assign-2', locale),
              '',
              i18n.get('help-page3-footer', locale),
              discord.empty,
            ].join('\n'),
          )
          .setFooter({ text: `${i18n.get('aliases', locale)}: /team, /p` }),
      ),
    new discord.Message()
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: i18n.get('help-page4-title', locale) })
          .setDescription([
            i18n.get('help-page4-negotiations', locale),
            '',
            i18n.get('help-page4-party-is-protected', locale),
            '',
            i18n.get('help-page4-inactive-users', locale),
            '',
            i18n.get(
              'help-page4-losing-protection',
              locale,
              PARTY_PROTECTION_PERIOD,
            ),
            '',
            i18n.get(
              'help-page4-steal-chance',
              locale,
              discord.emotes.smolStar,
            ),
            '',
            i18n.get(
              'help-page4-cooldown',
              locale,
              STEAL_COOLDOWN_HOURS / 24,
            ),
            '',
            i18n.get('help-page4-footer', locale),
            discord.empty,
          ].join('\n')),
      ),
    new discord.Message()
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: i18n.get('help-page10-title', locale) })
          .setDescription([
            i18n.get('help-page10-starting-points', locale),
            '',
            i18n.get('help-page10-stats-rng', locale),
            '',
            i18n.get('help-page10-attack', locale),
            '',
            i18n.get('help-page10-defense', locale),
            '',
            i18n.get('help-page10-speed', locale),
            '',
            i18n.get('help-page10-tower', locale),
            '',
            i18n.get('help-page10-reclear', locale),
            '',
            i18n.get('help-page10-distribution', locale),
            '',
            i18n.get('help-page10-keys', locale, RECHARGE_KEYS_MINS),
            discord.empty,
          ].join('\n')),
      ),
    new discord.Message()
      .addComponents([
        new discord.Component()
          .setId('buy', 'bguaranteed', userId, '5')
          .setLabel(`/buy guaranteed 5`),
      ])
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: i18n.get('help-page5-title', locale) })
          .setDescription([
            i18n.get('help-page5-shop', locale),
            '',
            i18n.get(
              'help-page5-daily-tokens',
              locale,
              RECHARGE_DAILY_TOKENS_HOURS,
            ),
            '',
            i18n.get('help-page5-buy-normal', locale, RECHARGE_MINS),
            '',
            i18n.get('help-page5-buy-guaranteed', locale),
            '',
            i18n.get('help-page5-example', locale, discord.emotes.smolStar),
            '',
            i18n.get(
              'help-page5-buy-3s',
              locale,
              discord.emotes.smolStar,
              COSTS.THREE,
            ),
            i18n.get(
              'help-page5-buy-4s',
              locale,
              discord.emotes.smolStar,
              COSTS.FOUR,
            ),
            i18n.get(
              'help-page5-buy-5s',
              locale,
              discord.emotes.smolStar,
              COSTS.FIVE,
            ),
            '',
            i18n.get('help-page5-footer', locale),
            discord.empty,
          ].join('\n'))
          .setFooter({ text: `${i18n.get('aliases', locale)}: /shop` }),
      ),
    new discord.Message()
      .addComponents([
        new discord.Component()
          .setLabel('GitHub')
          .setUrl('https://github.com/ker0olos/fable'),
        new discord.Component()
          .setLabel(i18n.get('support-server', locale))
          .setUrl('https://discord.gg/H69RVBxeYY'),
        new discord.Component()
          .setLabel(i18n.get('donate', locale))
          .setUrl('https://ko-fi.com/ker0olos'),
      ])
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: i18n.get('help-page6-title', locale) })
          .setDescription([
            i18n.get('help-page6-release', locale),
            '',
            '**[Classes And Skills](https://github.com/ker0olos/fable/issues/64)**',
            '',
            '**[Server vs. Server](https://github.com/ker0olos/fable/issues/65)**',
            '',
          ].join('\n')),
      ),
    new discord.Message()
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: i18n.get('help-page7-title', locale) })
          .setDescription([
            `- \`/gacha\` \`/w\`: _${i18n.get('/gacha', locale)}_`,
            `- \`/q\`: _${i18n.get('/q', locale)}_`,
            `- \`/pull\` \`/guaranteed\`: _${i18n.get('/pull', locale)}_`,
            '',
            `- \`/now\` \`/tu\`: _${i18n.get('/now', locale)}_`,
            `- \`/search\` \`/anime\` \`/manga\` \`/series\`: _${
              i18n.get('/search', locale)
            }_`,
            `- \`/character\` \`/char\`: _${i18n.get('/character', locale)}_`,
            '',
            `- \`/merge\` \`/synthesize\`: _${i18n.get('/merge', locale)}_`,
            '',
            `- \`/party view\` \`/team view\` \`/p view\`: _${
              i18n.get('/party view', locale)
            }_`,
            `- \`/party assign\` \`/team assign\` \`/p assign\`: _${
              i18n.get('/party assign', locale)
            }_`,
            `- \`/party swap\` \`/team swap\` \`/p swap\`: _${
              i18n.get('/party swap', locale)
            }_`,
            `- \`/party remove\` \`/team remove\` \`/p remove\`: _${
              i18n.get('/party remove', locale)
            }_`,
            '',
            `- \`/collection stars\` \`/coll stars\` \`/mm stars\`: _${
              i18n.get('/coll stars', locale)
            }_`,
            `- \`/collection media\` \`/coll media\` \`/mm media\`: _${
              i18n.get('/coll media', locale)
            }_`,
            '',
            `- \`/steal\`: _${i18n.get('/steal', locale)}_`,
            `- \`/trade\` \`/offer\`: _${i18n.get('/trade', locale)}_`,
            `- \`/give\` \`/gift\`: _${i18n.get('/give', locale)}_`,
            '',
            `- \`/battle tower\`: _${i18n.get('/battle tower', locale)}_`,
            `- \`/reclear\`: _${i18n.get('/reclear', locale)}_`,
          ].join('\n')),
      ),
    new discord.Message()
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: i18n.get('help-page8-title', locale) })
          .setDescription([
            `- \`/nick\`: _${i18n.get('/nick', locale)}_`,
            `- \`/image\` \`/custom\`: _${i18n.get('/image', locale)}_`,
            '',
            `- \`/found\` \`/owned\`: _${i18n.get('/found', locale)}_`,
            '',
            `- \`/likeslist\`: _${i18n.get('/likeslist', locale)}_`,
            '',
            `- \`/like\` \`/protect\` \`/wish\`: _${
              i18n.get('/like', locale)
            }_`,
            `- \`/unlike\`: _${i18n.get('/unlike', locale)}_`,
            `- \`/likeall\`: _${i18n.get('/likeall', locale)}_`,
            `- \`/unlikeall\`: _${i18n.get('/unlikeall', locale)}_`,
            '',
            `- \`/buy normal\` \`/shop normal\`: _${
              i18n.get('/buy normal', locale)
            }_`,
            `- \`/buy guaranteed\` \`/shop guaranteed\`: _${
              i18n.get('/buy guaranteed', locale)
            }_`,
            '',
            `- \`/automerge min\`: _${i18n.get('/automerge min', locale)}_`,
            `- \`/automerge max\`: _${i18n.get('/automerge max', locale)}_`,
            '',
            `- \`/installed packs\`: _${i18n.get('/installed packs', locale)}_`,
            '',
            `- \`/logs\`: _${i18n.get('/logs', locale)}_`,
          ].join('\n')),
      ),
    new discord.Message()
      .addEmbed(
        new discord.Embed().setAuthor({
          name: i18n.get('help-page9-title', locale),
        })
          .setDescription([
            i18n.get('community-packs-permission', locale),
            `- \`/packs install\`: _${i18n.get('/packs install', locale)}_`,
            `- \`/packs uninstall\`: _${i18n.get('/packs uninstall', locale)}_`,
            `- \`/packs disable builtins\`: _${
              i18n.get('/packs disable builtins', locale)
            }_`,
            '',
            i18n.get('make-your-own-pack', locale),
          ].join('\n')),
      ),
  ];

  return discord.Message.page({
    message: pages[index],
    total: pages.length,
    type: 'help',
    next: index + 1 < pages.length,
    index,
    locale,
  });
}

const help = {
  pages,
};

export default help;
