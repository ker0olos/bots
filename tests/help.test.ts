import { assertEquals } from '$std/assert/mod.ts';
import { assertMonochromeSnapshot } from '~/tests/utils.test.ts';

import help from '~/src/help.ts';

Deno.test('/help', async (test) => {
  await test.step('navigation', () => {
    const message = help.pages({ userId: 'user_id', index: 0 });

    assertEquals(message.json().data.components[0].components[0], {
      custom_id: 'help==8=prev',
      label: 'Prev',
      style: 2,
      type: 2,
    });

    assertEquals(message.json().data.components[0].components[1], {
      custom_id: '_',
      disabled: true,
      label: '1/9',
      style: 2,
      type: 2,
    });

    assertEquals(message.json().data.components[0].components[2], {
      custom_id: 'help==1=next',
      label: 'Next',
      style: 2,
      type: 2,
    });
  });

  await test.step('pages', async () => {
    for (let i = 0; i < 9; i++) {
      const message = help.pages({ userId: 'user_id', index: i });

      await assertMonochromeSnapshot(test, message.json());
    }
  });
});
