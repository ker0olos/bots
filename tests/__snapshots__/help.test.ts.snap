export const snapshot = {};

snapshot[`/help 1`] = `
{
  data: {
    attachments: [
    ],
    components: [
      {
        components: [
          {
            custom_id: "help==7=prev",
            label: "Prev",
            style: 2,
            type: 2,
          },
          {
            custom_id: "_",
            disabled: true,
            label: "1/8",
            style: 2,
            type: 2,
          },
          {
            custom_id: "help==1=next",
            label: "Next",
            style: 2,
            type: 2,
          },
          {
            custom_id: "gacha=user_id",
            label: "/gacha",
            style: 2,
            type: 2,
          },
          {
            custom_id: "now=user_id",
            label: "/now",
            style: 2,
            type: 2,
          },
        ],
        type: 1,
      },
    ],
    embeds: [
      {
        author: {
          name: "1.",
        },
        description: "Before anything: You will need characters.

__You get a free pull every 30 minutes (maxed at 5 pulls)__. Use \`/now\` to check how many pulls you have and how much time is left until your next free pull.

__Characters are exclusive__, once a player on this server finds a character, no one else can find that character in gacha again.

__Characters vary in ranks from 1<:smol_star:1088427421096751224>to 5<:smol_star:1088427421096751224>__ A character rating is based on their popularity (a character rating cannot be raised).

__Gacha rates depend on multiple variables__, but generally speaking the chances of a 5<:smol_star:1088427421096751224>appearing is equal or less than 1%.

> *\`/q\` allows you to do gacha pulls with no animations*
​",
        footer: {
          text: "aliases: /w, /q",
        },
        title: "\`/gacha\`",
        type: "rich",
      },
    ],
  },
  type: 4,
}
`;

snapshot[`/help 2`] = `
{
  data: {
    attachments: [
    ],
    components: [
      {
        components: [
          {
            custom_id: "help==0=prev",
            label: "Prev",
            style: 2,
            type: 2,
          },
          {
            custom_id: "_",
            disabled: true,
            label: "2/8",
            style: 2,
            type: 2,
          },
          {
            custom_id: "help==2=next",
            label: "Next",
            style: 2,
            type: 2,
          },
        ],
        type: 1,
      },
    ],
    embeds: [
      {
        author: {
          name: "2.",
        },
        description: '~~Synthesize Merge~~ Sacrifice characters to pull a new character with a higher rating.

__Be careful where you sacrifice your characters.__
Characters are the main currency in Fable. While synthesis returns a higher-rated character its *"currency"* value will always be lower than the sum of the sacrificed characters.

__Your party members and likes will never be sacrificed.__ To protect characters from synthesis, add them to your likeslist with \`/like\`

__It takes 5 characters from a specific rating__ to get a new character with 1 more star than the sacrificed.

__Fable will always auto-synthesize your lower-rated characters first.__

Example: \`/synthesize 3\` will sacrifice __25 of your 1<:smol_star:1088427421096751224>characters__
or __5 of your 2<:smol_star:1088427421096751224>characters__ or any other combination of both lower ratings.

> *Sacrificed characters will return to being available in \`/gacha\`*
​',
        footer: {
          text: "aliases: /merge",
        },
        title: "\`/synthesize\`",
        type: "rich",
      },
    ],
  },
  type: 4,
}
`;

snapshot[`/help 3`] = `
{
  data: {
    attachments: [
    ],
    components: [
      {
        components: [
          {
            custom_id: "help==1=prev",
            label: "Prev",
            style: 2,
            type: 2,
          },
          {
            custom_id: "_",
            disabled: true,
            label: "3/8",
            style: 2,
            type: 2,
          },
          {
            custom_id: "help==3=next",
            label: "Next",
            style: 2,
            type: 2,
          },
          {
            custom_id: "cstars=5=user_id",
            label: "/collection stars 5",
            style: 2,
            type: 2,
          },
          {
            custom_id: "cstars=4=user_id",
            label: "/collection stars 4",
            style: 2,
            type: 2,
          },
        ],
        type: 1,
      },
    ],
    embeds: [
      {
        author: {
          name: "3.",
        },
        description: "You will need a party to use combat features but all combat features are still in development, so parties are only an aesthetic right now.

__You can't have more than 1 party__ or save your current party to a preset and switch between presets.

__The more your party members are familiar with each other the better they preform__. This is called synergy, five party members from the same anime are better than 5 members from a mix of different media.

\`/party view\` to see your current party.

\`/party assign name: spot:\` to assign a character you have to your party;
leaving the \`spot:\` empty will assign the character to the first empty spot or override the last spot.

> *\`/collection\` can be used to browse your characters and select your potential party members*
​",
        footer: {
          text: "aliases: /team, /p",
        },
        title: "\`/party\`",
        type: "rich",
      },
    ],
  },
  type: 4,
}
`;

snapshot[`/help 4`] = `
{
  data: {
    attachments: [
    ],
    components: [
      {
        components: [
          {
            custom_id: "help==2=prev",
            label: "Prev",
            style: 2,
            type: 2,
          },
          {
            custom_id: "_",
            disabled: true,
            label: "4/8",
            style: 2,
            type: 2,
          },
          {
            custom_id: "help==4=next",
            label: "Next",
            style: 2,
            type: 2,
          },
          {
            custom_id: "now=user_id",
            label: "/vote",
            style: 2,
            type: 2,
          },
          {
            custom_id: "buy=bguaranteed=user_id=4",
            label: "/buy guaranteed 4",
            style: 2,
            type: 2,
          },
        ],
        type: 1,
      },
      {
        components: [
          {
            custom_id: "buy=bguaranteed=user_id=5",
            label: "/buy guaranteed 5",
            style: 2,
            type: 2,
          },
        ],
        type: 1,
      },
    ],
    embeds: [
      {
        author: {
          name: "Voting",
        },
        description: "Voting is an additional way to acquire character with a specific guaranteed rating.

__Voting on weekends__ awards **2** votes instead of **1** (_Saturday and Sunday_).

\`/buy random\` gets you random pulls, they are identical to the pulls you can get each 30 minutes.

\`/buy guaranteed\` however is special, while it can expansive it will guarantee that you get a character from one specific rank. Use the guaranteed pulls you bought by calling \`/pull stars:\`

> *Example: \`/buy guaranteed 5\` will guarantee you a random 5<:smol_star:1088427421096751224>character.*
​",
        footer: {
          text: "aliases: /shop",
        },
        type: "rich",
      },
    ],
  },
  type: 4,
}
`;

snapshot[`/help 5`] = `
{
  data: {
    attachments: [
    ],
    components: [
      {
        components: [
          {
            custom_id: "help==3=prev",
            label: "Prev",
            style: 2,
            type: 2,
          },
          {
            custom_id: "_",
            disabled: true,
            label: "5/8",
            style: 2,
            type: 2,
          },
          {
            custom_id: "help==5=next",
            label: "Next",
            style: 2,
            type: 2,
          },
          {
            label: "Full Roadmap",
            style: 5,
            type: 2,
            url: "https://github.com/ker0olos/fable/issues/1",
          },
          {
            label: "GitHub",
            style: 5,
            type: 2,
            url: "https://github.com/ker0olos/fable",
          },
        ],
        type: 1,
      },
      {
        components: [
          {
            label: "Discord Support Server",
            style: 5,
            type: 2,
            url: "https://discord.gg/H69RVBxeYY",
          },
          {
            label: "Donate",
            style: 5,
            type: 2,
            url: "https://github.com/sponsors/ker0olos",
          },
        ],
        type: 1,
      },
    ],
    embeds: [
      {
        author: {
          name: "Roadmap",
        },
        description: "__**Recently Added**__

\`/synthesize\`
_merge characters together to pull a new character_

__**Releasing in the near future (5 days ~ 2 months)**__

**[Daily Quests](https://github.com/ker0olos/fable/issues/75)**
Complete quests for rewards

**[Stealing](https://github.com/ker0olos/fable/issues/49)**
You will be able to steal characters from other users",
        type: "rich",
      },
    ],
  },
  type: 4,
}
`;

snapshot[`/help 6`] = `
{
  data: {
    attachments: [
    ],
    components: [
      {
        components: [
          {
            custom_id: "help==4=prev",
            label: "Prev",
            style: 2,
            type: 2,
          },
          {
            custom_id: "_",
            disabled: true,
            label: "6/8",
            style: 2,
            type: 2,
          },
          {
            custom_id: "help==6=next",
            label: "Next",
            style: 2,
            type: 2,
          },
        ],
        type: 1,
      },
    ],
    embeds: [
      {
        author: {
          name: "Essential Commands",
        },
        description: "- \`/gacha\` \`/w\`: _start a new gacha pull_
- \`/q\`: _start a gacha pull but with no animations_
- \`/pull\` \`/guaranteed\`: _use the guaranteed pulls you have_

- \`/now\` \`/vote\` \`/daily\` \`/tu\`: _check what you can do right now_
- \`/search\` \`/anime\` \`/manga\`: _search for specific media_
- \`/character\` \`/char\`: _search for a specific character_

- \`/synthesize\` \`/merge\`: _synthesize characters together to pull a new character_

- \`/party view\` \`/team view\` \`/p view\`: _view your current party_
- \`/party assign\` \`/team assign\` \`/p assign\`: _assign a character to your party_
- \`/party swap\` \`/team swap\` \`/p swap\`: _swap characters spots with each others_
- \`/party remove\` \`/team remove\` \`/p remove\`: _remove a character from your party_

- \`/collection list\` \`/coll list\` \`/mm list\`: _list user characters in bulks_
- \`/collection stars\` \`/coll stars\` \`/mm stars\`: _view user your stars_
- \`/collection media\` \`/coll media\` \`/mm media\`: _view user characters in a specific media_

- \`/trade\` \`/offer\`: _trade characters with another user_
- \`/give\` \`/gift\`: _give characters to another user_",
        type: "rich",
      },
    ],
  },
  type: 4,
}
`;

snapshot[`/help 7`] = `
{
  data: {
    attachments: [
    ],
    components: [
      {
        components: [
          {
            custom_id: "help==5=prev",
            label: "Prev",
            style: 2,
            type: 2,
          },
          {
            custom_id: "_",
            disabled: true,
            label: "7/8",
            style: 2,
            type: 2,
          },
          {
            custom_id: "help==7=next",
            label: "Next",
            style: 2,
            type: 2,
          },
        ],
        type: 1,
      },
    ],
    embeds: [
      {
        author: {
          name: "Other Commands",
        },
        description: "- \`/nick\`: _change the nickname of a character_
- \`/image\` \`/custom\`: _change the image of a character_

- \`/found\` \`/obtained\` \`/owned\`: _list all characters found from a specific media_

- \`/like\`: _like a character to be notified if someone finds them_
- \`/unlike\`: _remove character from your likes_
- \`/likeslist\`: _list user liked characters_

- \`/buy random\` \`/shop random\`: _use votes to buy random pulls_
- \`/buy guaranteed\` \`/shop guaranteed\`: _use votes to buy pulls with a specific guaranteed rating_

- \`/anilist next_episode\`: _find when the next episode of an anime is airing_",
        type: "rich",
      },
    ],
  },
  type: 4,
}
`;
